docs/enhancements/clusterpool-inventory.md

```diff
@@ -0,0 +1,175 @@
+# Clusterpool for on-prem cloud providers
+
+[HIVE-1576](https://issues.redhat.com/browse/HIVE-1367)
+
+- [ClusterPool Inventory](#clusterpool-inventory)
+    - [User Stories](#user-stories)
+    - [Problem Statement](#problem-statement)
+    - [Proposal](#proposal)
+        - [Summary](#summary)
+        - [`ClusterPool.Spec.Inventory`](#clusterpoolspecinventory)
+        - [How To Use](#how-to-use)
+        - [Validation](#validation)
+        - [`Size` and `MaxSize`](#size-and-maxsize)
+        - [Pool Version](#pool-version)
+        - [Handling Inventory Updates](#handling-inventory-updates)
+            - [Adding An Inventory](#adding-an-inventory)
+            - [Adding An Entry](#adding-an-entry)
+            - [Removing An Entry](#removing-an-entry)
+            - [Deleting The Inventory](#deleting-the-inventory)
+        - [Racing For Names](#racing-for-names)
+        - [Fairness](#fairness)
```

Comment:
TOC needs update. FWIW, I use the following VSCode extension, which keeps the TOC up to date automatically:

Name: Markdown All in One
Id: yzhang.markdown-all-in-one
Description: All you need to write Markdown (keyboard shortcuts, table of contents, auto preview and more)
Version: 3.4.0
Publisher: Yu Zhang
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one